contracts/tatum/Tatum721Provenance.sol

```diff
@@ -210,54 +183,55 @@ contract Tatum721Provenance is
             }
         }
     }
-    // function safeTransfer(
-    //     address to,
-    //     uint256 tokenId,
-    //     string memory data
-    // ) public payable {
-    //     Stringsutils.Slice memory s = str.toSlice(data);    
-    //     Stringsutils.Slice memory delim = str.toSlice("'''###'''");     
-    //     Stringsutils.Slice memory token;
-    //     // string[] memory parts = new string[](s.count(delim)+1);      
-    //     string[] memory parts = new string[](str.count(s,delim)+1);      
-    //     parts[0]=str.toString(str.split(s,delim,token));
-    //     parts[1]=str.toString(str.rsplit(s,delim,token));
-    //     uint value =_stringToUint(parts[1]);
-    //     bytes memory bytesData=abi.encodePacked(data);
+    function safeTransfer(
+        address to,
+        uint256 tokenId,
+        string calldata data
+    ) public payable {
+        uint index;
+        uint value;
+        bytes calldata dataBytes= bytes(data);
+        for(uint i=0;i<dataBytes.length;i++){
+            if(dataBytes[i]==0x27 && dataBytes[i+1]==0x27 && dataBytes[i+2]==0x27 && dataBytes[i+3]==0x23 && dataBytes[i+4]==0x23 && dataBytes[i+5]==0x23 && dataBytes[i+6]==0x27 && dataBytes[i+7]==0x27 && dataBytes[i+8]==0x27){
```

Comment:
check if the data length is in accessing [i+8] is actually pointing to the correct index - don't want to access array index which is not actually there.

Reply:
I still dont see the validation here. it should be like this
if(dataBytes[i]==0x27 && dataBytes.length > i+8 && ......)

Reply:
fix this!